source/common/http/http3/conn_pool.cc

```diff
@@ -109,7 +109,12 @@ void Http3ConnPoolImpl::onConnectFailed(Envoy::ConnectionPool::ActiveClient& cli
 }
 
 // Make sure all connections are torn down before quic_info_ is deleted.
-Http3ConnPoolImpl::~Http3ConnPoolImpl() { destructAllConnections(); }
+Http3ConnPoolImpl::~Http3ConnPoolImpl() {
+  destructAllConnections();
+  // set this to zero since state_ is not tracking connections for H3, only the initial capacity
```

Comment:
nit: Capitalize "Set" and end with "." to make this a complete sentence, as per the style guide.

https://google.github.io/styleguide/cppguide.html#Punctuation,_Spelling_and_Grammar

https://github.com/envoyproxy/envoy/pull/26107#discussion_r1137820717